flipperdevices/flipperzero-firmware
# [FL-2269] Core2 OTA

# What's new

- [FL-2269] Support for local updates of Core2 firmware 
- Added support for switching between different versions or feature sets of radio stacks
- Improved ble_glue & internal APIs related to C2 interaction
- Added utilities for parsing metadata of radio stack .bin images
- Cleaned up hardcoded stack metadata in all parts of scripts & makefiles
- Added FUS & stack details to debug output
- Added core2 mode info (FUS/stack) to `device_info` CLI command output
- Updated STM32Cube to 1.13.3
- Dropped hardware CRC usage and switched to software implementation
- Added Option Bytes (OB) verification to local updater 
- Added related scripts for converting `ob.data` into binary form suitable for storing in `update.fuf`
- Added OB self-healing in case of minor mismatches

# Verification 

- Manually remove radio stack or install an unsupported/deprecated one
- Build OTA package with `make COMPACT=1 DEBUG=0 updater_package` and install it
- Verify with `device_info` that radio stack is of latest version and fully functional.

# Checklist (For Reviewer)



[FL-2269]: https://flipperzero.atlassian.net/browse/FL-2269?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
